docs/source/examples.daisy_chain_gap.ipynb

```diff
@@ -0,0 +1,746 @@
+      "qqrr\n",
+      "Running wfl.generate.md._sample_autopara_wrappable with num_python_subprocesses=4, num_inputs_per_python_subprocess=1\n"
+     ]
+    }
+   ],
+   "source": [
+    "outputs = OutputSpec(\"2.ch.rdkit.md.traj.xyz\")\n",
+    "\n",
+    "md_params = {\n",
+    "    \"steps\": 80,\n",
+    "    \"dt\": 0.5,  # fs\n",
+    "    \"temperature\": 300,  # K\n",
+    "    \"temperature_tau\": 500,  \n",
+    "    \"results_prefix\": \"xtb_\",\n",
+    "    \"traj_step_interval\": 5}\n",
+    "\n",
+    "remote_info = {\n",
+    "    \"sys_name\" : \"local\", \n",
+    "    \"job_name\" : \"md\", \n",
+    "    \"resources\" : { \n",
+    "        \"max_time\" : \"15m\",\n",
+    "        \"num_cores\" : 4,\n",
+    "        \"partitions\" : \"any$\"}, \n",
+    "    \"partial_node\": True,\n",
+    "    \"check_interval\": 5,\n",
+    "    \"num_inputs_per_queued_job\" :20,\n",
+    "    \"pre_cmds\" :[\"conda activate dev\"]\n",
+    "}\n",
+    "\n",
+    "# set to None for github testing purposes\n",
+    "remote_info=None\n",
+    "\n",
+    "inputs = md.sample(\n",
+    "    inputs=inputs, \n",
+    "    outputs=outputs,\n",
+    "    calculator=xtb_calc,\n",
+    "    autopara_info = AutoparaInfo(\n",
+    "        remote_info=remote_info),\n",
+    "    **md_params\n",
+    "    )\n"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Filter out unreasonable structures\n",
+    "\n",
+    "While diverse training set leads to better model extrapolation, structures too dissimilar to the region of interest are fitted at the expense of accuracy elsewhere. One way to spot structures somewhat distant from equilibrium is by checking for high force components. Below we exclude such structures via one of wfl's filtering functions. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "\n",
+    "def are_xtb_force_comps_reasonable(at):\n",
+    "    force_comps = at.arrays[\"xtb_forces\"]\n",
+    "    return np.all(np.abs(force_comps) < 8 )\n",
+    "\n",
+    "outputs = OutputSpec(\"3.ch.rdkit.md.traj.filtered.xyz\") \n",
+    "inputs = by_bool_func(\n",
+    "    inputs = inputs, \n",
+    "    outputs = outputs,\n",
+    "    at_filter = are_xtb_force_comps_reasonable\n",
+    ")\n"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Calculate SOAP descriptor"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "tmp_outputs = OutputSpec()\n",
+    "outputs = OutputSpec(\"4.ch.rdkit.md.traj.filtered.soap.xyz\")\n",
+    "\n",
+    "# a nice feature of `ConfigSet`: \n",
+    "# can check if output from this step is already created and not re-execute it\n",
+    "if outputs.done():\n",
```

Comment:
FWIW, I personally wouldn't bother with this check in this case. I'd do the actual descriptor calculation to a file, which will be auto-skipped, and assume that the cost of the averaging operation is trivial.  Also, I thought I added a `map` function which will do fairly arbitrary operations _with autoparallelization_.

Reply:
Having that kind of example someplace, sure.  That it's needed in the example we're using for this paper I'm less convinced of.

Reply:
I agree, there's one already in the docs explaining how input/output abstraction works.